modules/web/src/app/settings/admin/defaults/static-labels-form/component.ts

```diff
@@ -0,0 +1,159 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Component, EventEmitter, Input, OnDestroy, OnInit, Output} from '@angular/core';
+import {FormArray, FormBuilder, FormGroup, Validators} from '@angular/forms';
+import {ResourceType} from '@app/shared/entity/common';
+import {EMPTY_STATIC_LABELS, StaticLabel} from '@app/shared/entity/settings';
+import {AsyncValidators} from '@app/shared/validators/async.validators';
+import {LabelFormValidators} from '@app/shared/validators/label-form.validators';
+import {Subject, takeUntil} from 'rxjs';
+
+enum Controls {
+  StaticLabels = 'staticLabels',
+  Key = 'key',
+  Values = 'values',
+  Protected = 'protected',
+  Default = 'default',
+}
+
+@Component({
+  selector: 'km-static-labels-form',
+  templateUrl: './template.html',
+  styleUrls: ['style.scss'],
+})
+export class StaticLabelsFormComponent implements OnInit, OnDestroy {
+  private _unsubscribe = new Subject<void>();
+  readonly Controls = Controls;
+
+  form: FormGroup;
+  @Input() staticLabels: StaticLabel[] = [];
+  asyncLabelValidators = [AsyncValidators.RestrictedLabelKeyName(ResourceType.Cluster)];
+  @Output() staticLabelsChange = new EventEmitter<StaticLabel[]>();
+
+  get statickLabelArray(): FormArray {
+    return this.form.get(Controls.StaticLabels) as FormArray;
+  }
+
+  constructor(private readonly _formBuilder: FormBuilder) {}
+
+  ngOnInit(): void {
+    this._initForm();
+
+    this.staticLabels?.forEach(label => {
+      this._addStaticLabel(label);
+    });
+    this._addStaticLabel(EMPTY_STATIC_LABELS);
+    this._updateLabelsObject();
+
+    this.form.valueChanges.pipe(takeUntil(this._unsubscribe)).subscribe(_ => {
+      this._updateLabelsObject();
+      this._addLabelIfNeeded();
+    });
+  }
+
+  ngOnDestroy(): void {
+    this._unsubscribe.next();
+    this._unsubscribe.complete();
+  }
+
+  deleteLabel(index: number): void {
+    this.statickLabelArray.removeAt(index);
+    this._updateLabelsObject();
+  }
+
+  isKeyUnique(index: number): void {
+    let duplications = 0;
+    const key = this.statickLabelArray.at(index).get(Controls.Key);
+
+    this.statickLabelArray.getRawValue().forEach((label: StaticLabel) => {
+      if (label.key === key.value) {
+        duplications++;
+      }
+    });
+
+    if (duplications > 1) {
+      key.setErrors({validLabelKeyUniqueness: true});
+    }
+  }
+
+  isProtected(val: boolean, index: number): boolean {
+    const defaultControl = this.statickLabelArray.at(index).get(Controls.Default);
+    if (val && defaultControl.value) {
+      this.statickLabelArray.at(index).get(Controls.Default).setValue(false);
+    }
+    return val;
+  }
```

Comment:
This method is semantically wrong, its returning the same value it got as input. If you want to modify the value of "Default" control, then please add a `change` event on the "Protected" checkbox e.g. `(change)="onProtectedValueChange($event, i)"`.
Also, shouldn't we set the value of "Default" to `true` when "Protected" is `true`?